src/services/WalletService/deterministic/useDeterministicWallet.ts

```diff
@@ -27,117 +40,152 @@ const useDeterministicWallet = (
     identity,
     'DeterministicWallet'
   );
-  const [shouldInit, setShouldInit] = useState(false);
-  const [service, setService] = useState<IDeterministicWalletService | undefined>(); // Keep a reference to the session in order to send
-  const [assetToQuery, setAssetToQuery] = useState(undefined as ExtendedAsset | undefined);
-  const [network, setNetwork] = useState(undefined as Network | undefined);
 
-  // Initialize DeterministicWallet and get the uri.
-  useEffect(() => {
-    if (!shouldInit || !assetToQuery || !network) return;
-    setShouldInit(false);
-    const dwService = DeterministicWalletService({
-      handleInitRequest: () =>
-        dispatch({
-          type: DWActionTypes.CONNECTION_REQUEST
-        }),
-      handleInit: (session: Wallet, asset: ExtendedAsset) =>
-        dispatch({
-          type: DWActionTypes.CONNECTION_SUCCESS,
-          payload: { session, asset }
-        }),
-      handleReject: (err: string) =>
-        dispatch({
-          type: DWActionTypes.CONNECTION_FAILURE,
-          error: {
-            code: DeterministicWalletReducer.errorCodes.SESSION_CONNECTION_FAILED,
-            message: err
-          }
-        }),
-      handleAccountsSuccess: () =>
-        dispatch({
-          type: DWActionTypes.GET_ADDRESSES_SUCCESS
-        }),
-      handleEnqueueAccounts: (accounts: DWAccountDisplay[]) => {
-        return dispatch({
-          type: DWActionTypes.ENQUEUE_ADDRESSES,
-          payload: { accounts }
+  const getAccounts = async (session: Wallet, dpaths: ExtendedDPath[]) => {
+    // Trezor wallet uses getMultipleAddresses for fetching multiple addresses at a time. Ledger doesn't have this functionality.
+    if (session.getMultipleAddresses) {
```

Comment:
Prefer defensive form:
`if (!session.getMultipleAddress) throw new Error()`

It saves an indentation level.